handlers/create.go

```diff
@@ -168,3 +184,113 @@ func buildMachineCreateCmd(machine *client.Machine) ([]string, error) {
 	log.Infof("Cmd slice: %v", cmd)
 	return cmd, nil
 }
+
+func createExtractedConfig(event *events.Event, machine *client.Machine) (string, error) {
+	// We are going to ignore doing anything for VirtualBox given that there is no way you can
+	// use Machine once it has been created.  Virtual is mainly a test-only use case
+	if strings.ToLower(machine.Driver) == "virtualbox" {
+		log.Debug("VirtualBox machine does not need config extracted")
+		return "", nil
+	}
+
+	// We will now zip, base64 encode the machine directory created, and upload this to cattle server.  This can be used to either recover
+	// the machine directory or used by Rancher users for their own local machine setup.
+	log.WithFields(log.Fields{
+		"resourceId": event.ResourceId,
+	}).Info("Creating and uploading extracted machine config")
+
+	// tar.gz the $CATTLE_HOME/machine/<machine-id>/machine/<machine-name> dir
+
+	// Open the source directory and read all the files we want to tar.gz
+	baseDir := filepath.Join(os.Getenv("CATTLE_HOME"), "machine", machine.ExternalId, "machine", "machines")
```

Comment:
We should keep the logic of determining and constructing the base machine path limited to one place.

Can you extract the similar logic from here: https://github.com/rancherio/go-machine-service/blob/master/handlers/create.go#L119-L123
And have one function that returns the basedMachineDIr (which you'll then have to join "machines" to for this use case.